### DISCUSSION TITLE
unset does not appear to work on Computed values which do not follow camel case naming convention.

### DISCUSSION THREAD

**Discussion by Geries** [livewire/livewire#8336] [1. Help]
### Issue

Non camel case compute variables cannot be unset.

### How to reproduce

In a livewire component put the following code:

```php
    #[Computed]
    protected function te_st()
    {
        dump('te_st compute');
        return 'test';
    }

    public function boot()
    {
        dump($this->te_st);
        unset($this->te_st);
        dump($this->te_st);
    }
```

The text `te_st compute` should appear only once on output, meaning the compute function only ran once.

### Solution

https://github.com/livewire/livewire/blame/1b1e0f02f2d23aaa8a5d7c8c977078aa3963206d/src/Features/SupportComputed/BaseComputed.php#L67

`if ($property !== $this->getName()) return;` should be I assume `if ($this->generatePropertyName($property) !== $this->getName()) return;` like with getter. I do not know if this fix affects anything else, this is how I'll fix it in my own application.

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147
